### PR TITLE
CUMULUS-1225: full list of config options (From #1014)

### PR DESCRIPTION
**Pull request by markdboyd**
Testing PR for forked PR #1014

### CODE REVIEW

File: docs/deployment/config_descriptions.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ hide_title: true
 
 ## Overview
 
-The table below provides an overview of the `config.yml` variables.
+The tables below provides overviews of the minimally required `config.yml` variables and all variables, respectively.
```
Comment (laurenfrederick) [inline docs/deployment/config_descriptions.md]:
```suggestion
The tables below provide overviews of the minimally required `config.yml` variables and all variables, respectively.
```

File: docs/deployment/config_descriptions.md [hunk 2]
```diff
@@ -32,7 +48,7 @@ dynamos:
 | cmr.provider | CUMULUS | the provider used for posting metadata to CMR
 | cmr.clientId | CUMULUS | the clientId used to authenticate with the CMR
 | cmr.password | (required) | the password used to authenticate with the CMR
-| buckets | (required) | Configuration of buckets with key, bucket name, and type (i.e. internal, public private)
+| buckets | (required) | Configuration of buckets with key, bucket name, and type (i.e. internal, public, private)
```
Comment (laurenfrederick) [inline docs/deployment/config_descriptions.md]: I would add protected here as well.

File: docs/deployment/README.md [hunk 3]
```diff
@@ -131,6 +131,10 @@ These buckets do not need any non-default permissions to function with Cumulus,
 
 ### VPC, Subnets and Security Group
 Cumulus supports operation within a VPC, but you will need to separately create the VPC, subnet, and security group for the Cumulus resources to use.
+You will need to set up VPC endpoints for the various services used by Cumulus if you wish to route traffic through the VPC.
```
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply docs/deployment/README.md]: Oh and add a note that this only needs to be done once per account.

File: docs/deployment/config_descriptions.md [hunk 4]
```diff
@@ -267,6 +300,8 @@ and 10 seconds.
 ### es
 Configuration for the Amazon Elasticsearch Service (ES) instance. Optional. Set `es: null` to disable ElasticSearch.
 
+VPC note: When using Amazon ElasticSearch Service in a VPC run `aws iam create-service-linked-role --aws-service-name es.amazonaws.com` before deploying.
```
Comment (laurenfrederick) [inline docs/deployment/config_descriptions.md]: Add a note that this only needs to be done once per account.

**Comment by markdboyd** [comment]
@laurenfrederick Ready for re-review

**Pull request merged**